src/DataSet/ObjectDataSet.php

```diff
@@ -13,106 +13,48 @@
 use Yiisoft\Validator\RulesProviderInterface;
 
 use function array_key_exists;
-use function get_class;
 
 /**
  * This data set makes use of attributes introduced in PHP 8. It simplifies rules configuration process, especially for
- * nested data and relations. Please refer to the guide for examples.
+ * nested data and relations. Please refer to the guide for example.
  *
  * @link https://www.php.net/manual/en/language.attributes.overview.php
  */
 final class ObjectDataSet implements RulesProviderInterface, DataSetInterface
 {
     private bool $dataSetProvided;
-    /**
-     * @var ReflectionProperty[]
-     */
-    private ?array $reflectionProperties = null;
-    private ?array $data = null;
+    private bool $rulesProvided;
 
-    private static array $cachedRules = [];
-    private string $cacheKey;
+    private ?array $cachedData = null;
```

Comment:
Maybe emphasize that is instance data. Also to avoid confusion with complex data in static cache.

Reply:
`$instanceDataCache` ?